File: go/1791.go
package main

func findCenter(edges [][]int) int {
	n := 0
	for i := 0; i < len(edges); i++ {
		if edges[i][0] > n {
			n = edges[i][0]
		}
		if edges[i][1] > n {
			n = edges[i][1]
		}
	}

	arr := make([]int, n+1)
	for i := 0; i < len(edges); i++ {
		arr[edges[i][0]]++
		arr[edges[i][1]]++
	}

	t := n - 1
	for i := 1; i <= len(arr); i++ {
		if arr[i] == t {
			return i
		}
	}

	return -1
}

File: cpp/1791.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int findCenter(vector<vector<int>> &edges) {
    int n = 0;
    for (int i = 0; i < edges.size(); i++) {
      int max = (edges[i][0] > edges[i][1]) ? edges[i][0] : edges[i][1];
      n = (max > n) ? max : n;
    }

    vector<int> arr(n + 1, 0);
    for (int i = 0; i < edges.size(); i++)
      ++arr[edges[i][0]], ++arr[edges[i][1]];

    int t = n - 1;
    for (int i = 1; i <= n; i++)
      if (arr[i] == t)
        return i;

    return -1;
  }
};
